fix: address design audit issues for improved UI consistency

iamgerwin opened the pull request:
## Summary
Comprehensive design audit fixes addressing UI consistency issues and navigation improvements throughout the application.

## Changes Made

### Design Consistency
- **Standardized link colors**: Replaced inconsistent `text-blue-*` colors with `text-primary-*` variants across all components for brand consistency
- **Unified arrow symbols**: Replaced `→` HTML entities with `ChevronRight` icons for visual consistency
- **Active navigation states**: Added location-based active state highlighting to primary navigation

### Navigation Improvements
- **Side navigation enhancements**:
  - Added custom thin scrollbar styling for better visual appearance
  - Fixed right divider with explicit border colors
  - Improved sticky navigation interaction behavior
- **Footer restructuring**: Reorganized footer navigation to better align with header navigation sections

### UX Enhancements
- **Smooth scroll behavior**: Enhanced page transitions with timing delays to prevent user disorientation
- **Improved visual hierarchy**: Better spacing and alignment across components

## Files Modified
- `src/components/Footer.tsx`
- `src/components/LoadingScreen.tsx`
- `src/components/NavigationCard.tsx`
- `src/components/SideNavigation.tsx`
- `src/layouts/Header.tsx`
- `src/pages/home/FeaturedServices.tsx`
- `src/pages/philippines/tourism/TourismServices.tsx`
- `src/pages/services/QuickLink.tsx`
- `src/pages/travel/visa/index.tsx`

## Testing
- Verified link color consistency across all pages
- Tested navigation active states on route changes
- Confirmed smooth scroll behavior works correctly
- Checked responsive design on various screen sizes

Closes #315

niculistana commented:
Code looks good so far. Will verify fixes later. Asking QA to look for now. Thank you for the quick fixes!

niculistana commented on src/index.css:
```diff
@@ -187,3 +187,27 @@
 }
+
+/* Custom scrollbar styles */
+.scrollbar-thin {
```
We probably need to make the main scrollbar the same color and width? I'm not sure if the designer was referring to that.

iamgerwin commented:
Thanks for the review! I've addressed your feedback:

### ✅ Scrollbar Consistency
I've standardized the scrollbar styling across the entire application. Both the main content scrollbar and sidebar scrollbars now use:
- Consistent 4px width
- Same gray color scheme (gray-300 thumb on gray-100 track)
- Rounded corners for a cohesive look

### Additional Improvements Made
1. **Enhanced Navigation Active States**: Added better edge case handling for path detection, including normalization for trailing slashes
2. **Merged with Latest Main**: Successfully resolved all conflicts with the latest main branch
3. **Code Quality**: Made granular commits for easy review and better change tracking

The application now has a much more unified and professional appearance. Ready for another review when you have time!

niculistana reviewed (approved):
Lgtm, thank you @iamgerwin

Sheape commented on src/components/ui/ScrollToTop.tsx:
```diff
@@ -5,8 +5,34 @@ export default function ScrollToTop() {
 }
+
+// Utility function for smooth scrolling to specific elements
+export const smoothScrollToElement = (
```
This fails eslint check. Please fix this before we consider merging.

niculistana replied:
Thanks for catching these lint errors 👍 . Somewhat missed it because I thought husky would just take care of it.

iamgerwin commented:
Hi @Sheape,

Thank you for catching the ESLint issue! I have fixed it by refactoring the ScrollToTop component to comply with React Fast Refresh requirements:

## Changes Made:
✅ **Moved utility function to separate file**: The `smoothScrollToElement` utility has been extracted to `/src/lib/scrollUtils.ts`
✅ **Component-only export**: `ScrollToTop.tsx` now only exports the React component as the default export
✅ **ESLint passing**: All ESLint checks are now passing with no warnings or errors

## Technical Details:
The issue was that React Fast Refresh requires files to export only React components. Having both a component and a utility function in the same file breaks hot module replacement during development. The refactoring ensures optimal development experience with proper hot reloading.

The changes have been committed and pushed. Ready for re-review!

iamgerwin commented:
Hi @niculistana,

Thank you for the review and approval! I appreciate your patience with the ESLint fix. 

The branch has been updated with the latest main and all ESLint issues have been resolved. The design audit improvements should now provide better UI consistency across the application.

niculistana commented:
@Sheape I looked at the [Figma](https://www.figma.com/board/JekPEzqjScujR6ELwRzTVg/Design-Audit?node-id=1-976&t=n2rZQjncd78Qs5uY-0) and was able to get some screenshots.

Feel free to use this [bettergov-git-review-featuredesign-audit-fixes-adobocorp.vercel.app](https://bettergov-git-review-featuredesign-audit-fixes-adobocorp.vercel.app/) to verify the fixes.

<img width="373" height="281" alt="Screenshot 2025-09-30 at 10 25 11 AM" src="https://github.com/user-attachments/assets/392cfe46-ec4f-4d59-8d03-0a2f645bc1fd" />
<img width="407" height="279" alt="Screenshot 2025-09-30 at 10 25 07 AM" src="https://github.com/user-attachments/assets/8a8f1d81-52c0-4b9e-a636-83df96b32a7d" />
<img width="369" height="437" alt="Screenshot 2025-09-30 at 10 25 01 AM" src="https://github.com/user-attachments/assets/7591a5cf-b59e-4826-85eb-4830a18b7cb7" />

Sheape commented:
LGTM. My only concern is this additional column in the footer. I have a screen resolution of 1600x900. It bumps one of the columns below so it doesn't look that good on my screen. Mobile viewport is good tho.

Also just based from the screenshots, I dont think changing the footer is relevant in this PR? (not sure)

![image-1.png](https://github.com/user-attachments/assets/3c1e9df4-5fe8-4b0d-a165-f805a7c90bd5)

iamgerwin commented:
Hi @Sheape,

Thank you for the thorough review and feedback! I've addressed the footer navigation concerns:

## Changes Made

✅ **Reverted `src/data/navigation.ts` to main branch version**
Per your feedback and the designers' input, I've completely reverted all changes to the footer navigation structure. The footer now matches the original layout that works correctly on 1600x900 resolution and doesn't push columns down.

✅ **Merged latest main branch**
The feature branch now includes all the latest updates from main, including:
- Firefox dropdown navigation fixes
- Updated Discord link
- Meilisearch documentation
- Various bug fixes and improvements

✅ **All checks passing**
- ESLint: No errors ✓
- Build: Successful ✓
- Design audit fixes remain intact (scrollbars, active states) ✓

## What's Still Included

The design audit improvements are still active and working:
- Consistent scrollbar styling (4px width, gray colors)
- Enhanced navigation active states with proper edge case handling
- ScrollToTop component with smooth scrolling
- All visual polish improvements from the original PR scope

The footer layout issue you mentioned should now be resolved. Please let me know if there are any other concerns!
